### PR TITLE
Add all Piston based Fixes from Carpet-Fixes

### PR DESCRIPTION
**Pull request by FxMorin**

### CODE REVIEW

File: src/main/java/ca/fxco/pistonlib/blocks/pistons/basePiston/BasicPistonBaseBlock.java [hunk 1]
```diff
@@ -240,13 +256,19 @@ public boolean triggerEvent(BlockState state, Level level, BlockPos pos, int typ
                     if (type != MotionType.PULL || frontState.isAir() ||
                             (frontState.getPistonPushReaction() != PushReaction.NORMAL && !frontState.is(ModTags.PISTONS)) ||
                             !canMoveBlock(frontState, level, frontPos, facing.getOpposite(), false, facing)) {
-                        level.removeBlock(headPos, false);
+                        if (!PistonLibConfig.illegalBreakingFix ||
+                                level.getBlockState(headPos).getDestroySpeed(level, headPos) != -1.0F) {
```
Comment (SpaceWalkerRS) [inline src/main/java/ca/fxco/pistonlib/blocks/pistons/basePiston/BasicPistonBaseBlock.java:295]: Would it make sense to check if the block is the appropriate head block instead? This implementation does allow breaking blocks with headless pistons, only preventing breaking blocks you couldn't also break by hand.

Comment (FxMorin) [reply src/main/java/ca/fxco/pistonlib/blocks/pistons/basePiston/BasicPistonBaseBlock.java:295]: Well there's also a headless piston fix. This is for people who want to use headless pistons but not being able to break blocks that cannot usually be broken by hand with them.

File: src/main/java/ca/fxco/pistonlib/mixin/fixes/PistonBaseBlock_pushOrderMixin.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import java.util.HashMap;
+
+/**
+ * Prevents pistons from being able to break blocks with a hardness value of -1.0F
+ * You need headless pistons to be able to break these blocks using pistons.
+ */
```
Comment (FxMorin) [inline src/main/java/ca/fxco/pistonlib/mixin/fixes/PistonBaseBlock_pushOrderMixin.java]:
```suggestion
/**
 * Fixes the piston update order being locational
 */
```
Oops wrong comment block

File: src/main/java/ca/fxco/pistonlib/blocks/pistons/basePiston/BasicPistonBaseBlock.java (hunk 1 above)
Comment (SpaceWalkerRS) [reply src/main/java/ca/fxco/pistonlib/blocks/pistons/basePiston/BasicPistonBaseBlock.java:295]: sounds good

**Review by SpaceWalkerRS — APPROVED** [review]

**Pull request merged**